[Bug] Unable to compile with the local protoc compiler

**Version of AimRT**
latest main branin with commit id 8b6283e68468a97c3010627a98f42964eafaa0ed

**Environments**
gcc 11.4
cmake version 3.28.3
OS: Ubuntu 22.04.5 LTS x86_64
libprotoc 26.1

**Describe the bug**
Unable to compile the project using the local protoc

**To Reproduce**
Steps to reproduce the behavior:
1. under aimrt root directory, just run

```bash
./build.sh -DAIMRT_USE_LOCAL_PROTOC_COMPILER=ON
```

**Expected behavior**
Compilation successful.

**Actual behavior**
编译报错，具体错误信息如下：

```
gmake[2]: *** No rule to make target 'protoc', needed by 'src/protocols/common/module.pb.cc'.  Stop.
gmake[2]: *** No rule to make target 'protoc', needed by 'src/protocols/example/rpc.pb.cc'.  Stop.
gmake[2]: *** No rule to make target 'protoc', needed by 'src/protocols/plugins/time_manipulator_plugin/time_manipulator.pb.cc'.  Stop.
gmake[2]: *** No rule to make target 'protoc', needed by 'src/protocols/plugins/parameter_plugin/parameter.pb.cc'.  Stop.
gmake[2]: *** No rule to make target 'protoc', needed by 'src/protocols/plugins/record_playback_plugin/record_playback.pb.cc'.  Stop.
gmake[2]: *** No rule to make target 'protoc', needed by 'src/protocols/plugins/log_control_plugin/log_control.pb.cc'.  Stop.
gmake[1]: *** [CMakeFiles/Makefile2:6911: src/protocols/common/CMakeFiles/aimrt_protocols_common_pb_gencode.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
gmake[1]: *** [CMakeFiles/Makefile2:6964: src/protocols/example/CMakeFiles/aimrt_protocols_example_pb_gencode.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:7018: src/protocols/plugins/time_manipulator_plugin/CMakeFiles/aimrt_protocols_time_manipulator_plugin_pb_gencode.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:7072: src/protocols/plugins/parameter_plugin/CMakeFiles/aimrt_protocols_parameter_plugin_pb_gencode.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:7126: src/protocols/plugins/record_playback_plugin/CMakeFiles/aimrt_protocols_record_playback_plugin_pb_gencode.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:7180: src/protocols/plugins/log_control_plugin/CMakeFiles/aimrt_protocols_log_control_plugin_pb_gencode.dir/all] Error 2
```

编译日志如下文件：
[build.log](https://github.com/user-attachments/files/17660762/build.log)

**Additional context**
Add any other context about the problem here.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -178,6 +178,7 @@ if(AIMRT_BUILD_WITH_ROS2)
       OFF
       CACHE BOOL "")
   include(GetJsonCpp)
+  include(GetYamlCpp)
   find_package(rclcpp REQUIRED)
   find_package(Python3 REQUIRED)
 endif()

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -173,10 +173,6 @@ if(AIMRT_BUILD_WITH_ROS2)
     cmake_policy(SET CMP0148 OLD)
   endif()
 
-  # fix gflag
-  set(BUILD_TESTING
-      OFF
-      CACHE BOOL "")
   include(GetJsonCpp)
   include(GetYamlCpp)
   find_package(rclcpp REQUIRED)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -148,9 +148,15 @@ if(AIMRT_BUILD_WITH_PROTOBUF)
   include(ProtobufGenCode)
 
   if(AIMRT_USE_LOCAL_PROTOC_COMPILER)
+    find_program(PROTOC_EXECUTABLE protoc)
+    if(NOT PROTOC_EXECUTABLE)
+      message(FATAL_ERROR "AIMRT_USE_LOCAL_PROTOC_COMPILER is ON, but can not find protoc compiler, " #
+                          "please install protoc and make it available in your PATH.")
+    endif()
+    message(STATUS "Found local protoc compiler: ${PROTOC_EXECUTABLE}")
     set(Protobuf_PROTOC_EXECUTABLE
-        "protoc"
-        CACHE STRING "Path to protoc compiler.")
+        ${PROTOC_EXECUTABLE}
+        CACHE STRING "Path to local protoc compiler.")
     add_executable(aimrt::protoc IMPORTED GLOBAL)
     set_target_properties(aimrt::protoc PROPERTIES IMPORTED_LOCATION ${Protobuf_PROTOC_EXECUTABLE})
     set_property(GLOBAL PROPERTY PROTOC_NAMESPACE_PROPERTY "aimrt")

